ca/server.go

```diff
@@ -530,13 +534,8 @@ func (s *Server) evaluateAndSignNodeCert(ctx context.Context, node *api.Node) {
 
 // signNodeCert does the bulk of the work for signing a certificate
 func (s *Server) signNodeCert(ctx context.Context, node *api.Node) {
-	if !s.securityConfig.RootCA().CanSign() {
-		log.G(ctx).WithFields(logrus.Fields{
-			"node.id": node.ID,
-			"method":  "(*Server).signNodeCert",
-		}).Errorf("no valid signer found")
-		return
-	}
+	rootCA := s.securityConfig.RootCA()
+	externalCA := s.securityConfig.ExternalCA()
```

Comment:
Nit: Right now, every call to `signNodeCert` creates a new `ExternalCA`, even if there are no external CA URLs configured. What if the `SecurityConfig` contained a pointer to an `ExternalCA` if appropriate, or `nil` if there are no URLs? Then `ExternalCA` would not need a `CanSign` method, because the presence of an `ExternalCA` object would imply that there's an external CA available.